### PR TITLE
Extract ActionView from Rails

### PR DESCRIPTION
**Pull request by delner**
As a next step towards simplifying the Rails integration, this pull request extracts all the ActionView instrumentation from the Rails integration into a new `action_view` integration. This new integration is automatically activated by the Rails integration, so users with `use :rails` should see no change.

Changes in this PR are dependent on #747. After that's merged, rebase and merge this to the dev branch.

### CODE REVIEW

File: lib/ddtrace/contrib/action_view/instrumentation.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require 'ddtrace/contrib/action_view/ext'
+
+module Datadog
+  module Contrib
+    module ActionView
+      # Defines instrumentation for ActionView
+      module Instrumentation
+        # TODO: Add instrumentation here.
```
Comment (brettlangdon) [inline lib/ddtrace/contrib/action_view/instrumentation.rb]: PR still WIP?

**Comment by delner** [comment]
Reworked this a little bit:

 - Rebased on the latest changes for `0.27-dev` (removes 1.9 support)
 - Changed the patching strategy from rewrite to prepend for Rails 3.1+

Might want to consider adding some tests for using ActionView in a standalone way in the future, but for now the existing Rails tests should be sufficient. Should be ready for review @brettlangdon

File: lib/ddtrace/contrib/action_view/configuration/settings.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        class Settings < Contrib::Configuration::Settings
+          option  :analytics_enabled,
+                  default: -> { env_to_bool(Ext::ENV_ANALYTICS_ENABLED, false) },
+                  lazy: true
+
+          option  :analytics_sample_rate,
+                  default: -> { env_to_float(Ext::ENV_ANALYTICS_SAMPLE_RATE, 1.0) },
+                  lazy: true
+
+          option :service_name, default: Ext::SERVICE_NAME
```
Comment (brettlangdon) [inline lib/ddtrace/contrib/action_view/configuration/settings.rb]: Are there cases when `action_view` will be the top level service?

I wonder if we need to even set a service name at all? Should this just be grouped under the Rails service? or should we have as something like:

```ruby
option :service_name, default: ENV['DATADOG_SERVICE_NAME'] ? "#{ENV['DATADOG_SERVICE_NAME']}-#{Ext::SERVICE_NAME}" : Ext::SERVICE_NAME
```

File: lib/ddtrace/contrib/action_view/instrumentation.rb [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+          # Rails < 3.1 template rendering
+          module Rails30
+            # rubocop:disable Metrics/MethodLength
+            def self.prepended(base)
+              # rubocop:disable Metrics/BlockLength
+              base.class_eval do
+                def render_with_datadog(*args, &block)
+                  # NOTE: This check exists purely for Rails 3.0 compatibility.
+                  #       The 'if' part can be removed when support for Rails 3.0 is removed.
+                  if active_datadog_span
```
Comment (brettlangdon) [inline lib/ddtrace/contrib/action_view/instrumentation.rb]: Not sure I fully follow this logic. I am sure it makes sense, just I think I am missing something.

Comment (brettlangdon) [reply lib/ddtrace/contrib/action_view/instrumentation.rb]: e.g. why do we need to do this?

File: lib/ddtrace/contrib/action_view/utils.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+module Datadog
+  module Contrib
+    module ActionView
+      # common utilities for Rails
+      module Utils
+        module_function
+
+        # in Rails the template name includes the template full path
+        # and it's better to avoid storing such information. This method
```
Comment (brettlangdon) [inline lib/ddtrace/contrib/action_view/utils.rb]: Why? It should be ok, no?

File: docs/GettingStarted.md [hunk 5]
```diff
@@ -345,11 +347,33 @@ For a list of available integrations, and their configuration options, please re
...
+```
+
+Where `options` is an optional `Hash` that accepts the following parameters:
+
+| Key | Description | Default |
+| ---| --- | --- |
+| `analytics_enabled` | Enable analytics for spans produced by this integration. `true` for on, `nil` to defer to global setting, `false` for off. | `false` |
+| `service_name` | Service name used for rendering instrumentation. | `action_view` |
+| `tracer` | `Datadog::Tracer` used to perform instrumentation. Usually you don't need to set this. | `Datadog.tracer` |
+| `template_base_path` | Used when the template name is parsed. If you don't store your templates in the `views/` folder, you may need to change this value | `'views/'` |
```
[the remark this answers is not in the document]
Comment (brettlangdon) [reply docs/GettingStarted.md]: Interesting question, we might be able to, but can defer for the future.

File: lib/ddtrace/contrib/action_view/configuration/settings.rb (hunk 2 above)
Comment (brettlangdon) [reply lib/ddtrace/contrib/action_view/configuration/settings.rb]: We'll get this default or the Rails service name by default.

**Pull request merged**